external/netutils/netlib_netmon.c

```diff
@@ -0,0 +1,173 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#if defined(CONFIG_NET_NETMON) && CONFIG_NSOCKET_DESCRIPTORS > 0
+
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include <netinet/in.h>
+#include <net/if.h>
+
+#include <netutils/netlib.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_NET_IPv4)
+#define PF_INETX PF_INET
+#elif defined(CONFIG_NET_IPv6)
+#define PF_INETX PF_INET6
+#endif
+#define NUM_SOCKETS MEMP_NUM_NETCONN
+
+/**
+ * Print a externally used socket information.
+ */
+static void print_socket(FAR struct socket sock[])
+{
+	int i;
+	int ret_chk;
+	char get_name[32];
+
+	printf("\nSockfd\tProto\tType\tLocal\t\t\tRemote\t\t\tstate\tPID\n");
+
+	for (i = 0; i < NUM_SOCKETS; ++i) {
+	/* Protect socket array */
+		if (sock[i].conn) {
+			printf("%-8d",i);
+			/* Protocol */
+			if (sock[i].conn->type & NETCONN_TCP) {
+				printf("TCP\t");
+#ifdef CONFIG_NET_IPv6
+				if (sock[i].conn->type & NETCONN_TYPE_IPV6) {
+					printf("IPv6\t");
+				} else {
+					printf("IPv4\t");
+				}
+#else
+				printf("IPv4\t");
+#endif
+				printf("%-15s:%-8s", inet_ntoa(sock[i].conn->pcb.ip->local_ip),"*");
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->remote_ip), sock[i].conn->pcb.tcp->remote_port);
+			} else if (sock[i].conn->type & NETCONN_UDP) {
+				printf("UDP\t");
+#ifdef CONFIG_NET_IPv6
+				if (sock[i].conn->type & NETCONN_TYPE_IPV6) {
+					printf("IPv6\t");
+				} else {
+					printf("IPv4\t");
+				}
+#else
+				printf("IPv4\t");
+#endif
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->local_ip),sock[i].conn->pcb.udp->local_port);
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->remote_ip), sock[i].conn->pcb.udp->remote_port);
+			} else {
+				printf("RAW\t");
+#ifdef CONFIG_NET_IPv6
```

Comment:
Duplcated code comes 3times here, I think these can be changed to MACRO or function.

Reply:
That's good point. Thanks for your comments. Macro is included for now.